cmd/cli/app/root.go

```diff
@@ -88,11 +109,15 @@ func init() {
 		RootCmd.Printf("error: %s", err)
 		os.Exit(1)
 	}
+
+	RootCmd.AddCommand(configHelpCmd)
 }
 
 func initConfig() {
 	if cfgFile != "" {
 		viper.SetConfigFile(cfgFile)
+	} else if os.Getenv("MINDER_CONFIG") != "" {
+		viper.SetConfigFile(os.Getenv("MINDER_CONFIG"))
```

Comment:
I was thinking this should already be provided by cobra/viper via the Bind and AutomaticEnv methods. This works too though 👍

Reply:
I've quickly tried this locally and seemed to work -
```diff
diff --git a/cmd/cli/app/root.go b/cmd/cli/app/root.go
index 80feec33..a6f0cb52 100644
--- a/cmd/cli/app/root.go
+++ b/cmd/cli/app/root.go
@@ -18,6 +18,7 @@ package app
 
 import (
 	"os"
+	"strings"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
@@ -28,8 +29,6 @@ import (
 )
 
 var (
-	cfgFile string // config file (default is $PWD/config.yaml)
-
 	// RootCmd represents the base command when called without any subcommands
 	RootCmd = &cobra.Command{
 		Use:   "minder",
@@ -65,9 +64,13 @@ func init() {
 	RootCmd.PersistentFlags().Bool("grpc-insecure", false, "Allow establishing insecure connections")
 	RootCmd.PersistentFlags().String("identity-url", constants.IdentitySeverURL, "Identity server issuer URL")
 	RootCmd.PersistentFlags().String("identity-client", "minder-cli", "Identity server client ID")
-	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "Config file (default is $PWD/config.yaml)")
+	RootCmd.PersistentFlags().String("config", "", "Config file (default is $PWD/config.yaml)")
 
 	// Bind viper config to cobra flags
+	if err := viper.BindPFlag("config", RootCmd.PersistentFlags().Lookup("config")); err != nil {
+		RootCmd.Printf("error: %s", err)
+		os.Exit(1)
+	}
 	if err := viper.BindPFlag("grpc_server.host", RootCmd.PersistentFlags().Lookup("grpc-host")); err != nil {
 		RootCmd.Printf("error: %s", err)
 		os.Exit(1)
@@ -88,9 +91,14 @@ func init() {
 		RootCmd.Printf("error: %s", err)
 		os.Exit(1)
 	}
+	viper.AutomaticEnv()
 }
 
 func initConfig() {
+	viper.SetEnvPrefix("minder")
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+
+	cfgFile := viper.GetString("config")
 	if cfgFile != "" {
 		viper.SetConfigFile(cfgFile)
 	} else {
```

Reply:
I've done this, though TBH it seems a lot more magical.  😁